File: Intermed/functionArr.py
def process_array(arr):
    arr[0] = 0
    arr[1] = 0

def main():
    SIZE = 5
    my_array = [1, 2, 3, 4, 5]

    for num in my_array:
        print(num, end=" ")

    # Passing the array to the function
    process_array(my_array)

    print()  # Print a newline

    for num in my_array:
        print(num, end=" ")

if __name__ == "__main__":
    main()

File: Intermed/functionArr.cpp
#include <iostream>

// Function to process the array
void processArray(const int arr[]) {
    
    arr[0] = 0;
    arr[1] = 0;
    
}

int main() {
    const int SIZE = 5;
    int myArray[SIZE] = {1, 2, 3, 4, 5};

    for (int i = 0; i < SIZE; ++i) {
        std::cout << myArray[i] << " ";
    }
    //prints 1 2 3 4 5
    // Passing the array to the function
    processArray(myArray);

    for (int i = 0; i < SIZE; ++i) {
        std::cout << myArray[i] << " ";
    }
    //print 0 0 3 4 5
    return 0;
}
